pkg/cloud/services/awsnode/cni.go

```diff
@@ -67,6 +63,41 @@ func (s *Service) ReconcileCNI(ctx context.Context) error {
 		return ErrCNIMissing
 	}
 
+	envVars := s.scope.AWSNodeEnvVars()
+	if len(envVars) > 0 {
+		s.scope.Info("updating aws-node daemonset environment variables", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
+
+		for i := range ds.Spec.Template.Spec.Containers {
+			container := &ds.Spec.Template.Spec.Containers[i]
+			if container.Name == "aws-node" {
+				existingVars := s.filterEnv(container.Env)
+
+				for ei, e := range existingVars {
+					for ai, a := range envVars {
+						if e.Name == a.Name {
+							existingVars[ei].Value = a.Value
+
+							envVars = append(envVars[:ai], envVars[ai+1:]...)
+							break
+						}
+					}
+				}
+				container.Env = append(existingVars, envVars...)
+				break
+			}
+		}
+		if s.scope.SecondaryCidrBlock() == nil {
+			err = remoteClient.Update(ctx, &ds, &client.UpdateOptions{})
+			if err != nil {
+				return err
+			}
```

Comment:
If the secondary cidr block is nil we could return here instead of doing the check again outside this block.